### ISSUE TITLE
Clarity for scale factor for custom dataset

### ISSUE THREAD

**Issue by poornimajd** [nianticlabs/monodepth2#232]
Hi,
I have trained the model using mono-stereo setting.While testing it using --eval_mono (testing on a single image) ,to get the distance in real world, I did the following
I used disparity value i.e disp_resized_np from this line
https://github.com/nianticlabs/monodepth2/blob/master/test_simple.py#L137

Depth=(focal length * baseline)/disparity value   (equation 1)
Once I got the depth, I used a method similar to median scaling to figure out the scale factor and then multiplied this with the above estimated depth ,to get the final depth.
Is this the correct way to final depth,or am I not suppose to use any scale factor apart from equation 1?

I am sorry to raise this question though there are a few issues similar to it(I went through most of them),but I could not find the precise answer to my question.

Basically I want to know if equation 1 is my final depth , or the scaling factor is still necessary.
Any suggestion is greatly appreciated!
Thanks in advance

**Comment by poornimajd** [comment]
Hi,
I just wanted to bring this up again.I did go through issues related to this,but I am not still clear with scale factor.I did not find a precise answer yet.
Any suggestion is helpful.
Thank you

**Comment by mrharicot** [comment]
Hi,
The disparity you want to use is `scaled_disp` as it corresponds to a (un)scaled inverse depth.
Even better you can modify the code to get the inverse depth:
```python
scaled_disp, scaled_depth = disp_to_depth(disp, 0.1, 100)
```
This is the depthmap you need to find the single scalar value (via median scaling for instance) to turn it into meters.

I hope this helps!

**Comment by poornimajd** [comment]
Thanks for the reply!


> ```python
> scaled_disp, scaled_depth = disp_to_depth(disp, 0.1, 100)
> ```

I tried distance(real world units)=scaled_depth*scale factor
but when I did distance=((baseline*focal_length)/scaled_disp)*scale_factor,I got a reasonable answer.So we need to take even the baseline and focallength rather than , depth=1/disparity right?

**Comment by mrharicot** [comment]
Oh my bad you trained using a stereo dataset!
Then in this case you can simply scale the scaled depth using the ratio between the baseline used by monodepth2: 0.1 and the real baseline in meters.
In the case of kitti the baseline is 0.54m so the ratio is 5.4.
